app/controllers/admin/enterprises_controller.rb

```diff
@@ -49,8 +49,11 @@ def edit
       @object.build_custom_tab if @object.custom_tab.nil?
       return unless params[:stimulus]
 
-      @enterprise.is_primary_producer = params[:is_primary_producer]
-      @enterprise.sells = params[:enterprise_sells]
+      if params[:is_primary_producer].present?
+        @enterprise.is_primary_producer = params[:is_primary_producer]
+      end
+      @enterprise.sells = params[:enterprise_sells] if params[:enterprise_sells].present?
```

Comment:
As the values can come from 2 different sides (Primary Details tab or Admin only tab), we need to control the presence of the value first.